python/xgrammar/matcher.py

```diff
@@ -121,20 +126,49 @@ def apply_token_bitmask_inplace(
         apply the bitmask to all logits in the batch.
     """
     if bitmask.device != logits.device:
-        raise ValueError(
-            "logits and bitmask should be on the same device. "
-            + f"But got logits.device: {logits.device}, bitmask.device: {bitmask.device}"
+        raise DeviceValidationError(
+            DeviceValidationError.device_mismatch(logits.device, bitmask.device)
         )
 
     if logits.device.type == "cuda":
-        if os.environ.get("XGRAMMAR_TOKEN_BITMASK_TRITON") == "1":
-            apply_token_bitmask_inplace_triton(logits, bitmask, indices)
+        if (
+            os.environ.get("XGRAMMAR_TOKEN_BITMASK_TRITON") == "1"
+            and "triton" in apply_token_bitmask_inplace_impl
```

Comment:
this logic will also fall back to cuda if triton isn't installed even if you explicitly set XGRAMMAR_TOKEN_BITMASK_TRITON. Alternative we could throw an error.

Reply:
We can throw an error for this. I have done that in the new commit.